### DISCUSSION TITLE
Performance question

### DISCUSSION THREAD

**Discussion by McFSR** [wlcrs/huawei_solar#178] [General]
Hi

I am currently using a heavily modified version of this integration https://github.com/Emilv2/huawei_solar.
I had to modify it because it had huge performance issues, making it nearly impossible to refresh all sensors once per minute. This is a known issue with that integration.

Your integration seems very mature and complete, great work! 

I can see it also relies on Emilv2's base library to communicate with the Sun2000 (at least that's what I can see in the manifest). So my question is: Has this integration problems to refresh all sensors/values once per minute (or once per 30 seconds) or are these performance issues gone in your solution?

Thanks a lot!

**Comment by wlcrs** [comment]
I've reworked that library extensively. A full changelog is here: https://gitlab.com/Emilv2/huawei-solar/-/compare/v1...master?from_project_id=17089010&straight=false

A full update of all sensors is now achieved by 4 modbus read-requests (in which I batch reading multiple modbus registers) instead of 50+ separate read-requests of 1 modbus register at a time.

**Reply by McFSR** [reply]
Wow, great work! Cannot wait to see it in action :-)

Tiny question: Is there any specific reason why the refresh interval of 30s did not make it into the configuration? I'm new to Home Assistant development, so I don't know if that is something rather complicated to achieve.

Thanks!

**Comment by RabAusten** [comment]
> Hi
> 
> I am currently using a heavily modified version of this integration https://github.com/Emilv2/huawei_solar. I had to modify it because it had huge performance issues, making it nearly impossible to refresh all sensors once per minute. This is a known issue with that integration.
> 
> Your integration seems very mature and complete, great work!
> 
> I can see it also relies on Emilv2's base library to communicate with the Sun2000 (at least that's what I can see in the manifest). So my question is: Has this integration problems to refresh all sensors/values once per minute (or once per 30 seconds) or are these performance issues gone in your solution?
> 
> Thanks a lot!

I've been using the integration for over a week and I am updating every 10 seconds with no issues. I have an automation to poll every 10 secs during the day, then only every minute during the night. Works great.
